File: rust/1672-richest-customer-wealth/src/main.rs
const DATA: [[u32; 3]; 3] = [[7, 1, 3], [2, 8, 7], [1, 9, 5]];

fn main() {
    if DATA.len() < 1 {
        panic!("No data given");
    } else if DATA.len() > 100 {
        panic!("Given data is too big");
    }

    let richest: Option<u32> = DATA
        .iter()
        .map(|account| {
            if account.len() <= 50 {
                account.iter().sum()
            } else {
                0
            }
        })
        .max();
    println!(
        "{:?}",
        richest.expect("couldn't find the richest customer wealth")
    );
}

File: go/1672-richest-customer-wealth/main.go
package main

import (
	"math"
)

func GetHighestWealth(data [][]uint) uint {
	//var data = [3][3]uint{{2, 8, 7}, {7, 1, 3}, {1, 9, 5}}
	var highest uint

	for _, customer := range data {
		var wealth uint
		for _, account := range customer {
			wealth += account
		}
		highest = uint(math.Max(float64(highest), float64(wealth)))
	}

	return highest
}
